### PR TITLE
Make web examples runnable with latest dart

### PR DESCRIPTION
**Pull request by dlchet**
- pubspec gets latest build_web_compilers
- README.md specifies working steps to run web examples
- minor linting issues are ignored

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ environment:
 
 dev_dependencies:
   build_runner: ^2.1.2
-  build_web_compilers: ^3.0.0
-  lints: ^1.0.1
+  build_web_compilers: ^4.0.9
+  lints: ^2.0.0
```
Comment (hoc081098) [inline pubspec.yaml:21]: We should keep the old version of 2 deps, because we support dart >= 2.12

Comment (dlchet) [reply pubspec.yaml:21]: got it. i'll see if i can rework this so the example can be run on 2.12 as well as latest.

Comment (dlchet) [reply pubspec.yaml:21]: well, i'm not sure how to make the examples work with past versions. i'm on dart 3.3.1 and i run into issues both with the old style of building
```
% dart run build_runner serve example
[SEVERE] build_web_compilers:sdk_js on asset:build_web_compilers/$package$ (cached):
Error compiling dartdevc module:build_web_compilers|lib/src/dev_compiler/dart_sdk.sound.js
Unhandled exception:
PathNotFoundException: Cannot open file, path = '/opt/homebrew/Caskroom/flutter/3.3.3/flutter/bin/cache/dart-sdk/lib/_internal/ddc_platform_sound.dill' (OS Error: No such file or directory, errno = 2)
```
and the new style
```
% webdev serve example               
[SEVERE] Pubspec errors: [The `build_web_compilers` version – 3.2.7 – is not within the allowed constraint – ^4.0.4.
null]
webdev could not run for this project.
The `build_web_compilers` version – 3.2.7 – is not within the allowed constraint – ^4.0.4.
```

So I'm a bit perplexed on what if anything should be changed. It seems to me that this problem would surface when using a recent Dart even in a separate project, so maybe I will try to make a minimal reproduction in a separate project, and then submit as an issue? @hoc081098 thank you for your attention, let me know if you see anything I'm missing

Comment (hoc081098) [reply pubspec.yaml:21]: @dlchet I think we can use a separate `pubspec.yaml` for web example (similar to `flutter/github_search`).
Could you try this approach?

**Comment by hoc081098** [comment]
Thanks, but it is obsolete now, because I have converted web example to a new project (by using melos), and it can use latest version of dependencies

**Pull request closed without merging**